pyccel/stdlib/math/pyc_math.h

```diff
@@ -8,6 +8,14 @@
 
 #include <stdint.h>
 
+/*
+** (((N % M) + M) % M) and fmod(fmod(N, M) + M, M)) are used to handle the negative
+** operands of modulo operator.
+*/
+
+#define MOD_PYC(N, M) ((N < 0 || M < 0) ? (((N % M) + M) % M) : (N % M))
+#define FMOD_PYC(N, M) ((N < 0 || M < 0) ? fmod(fmod(N, M) + M, M) : fmod(N, M))
```

Comment:
> Shouldn't this be an XOR?

it can be both.

Reply:
> Are you sure the second modulo is needed? Or is it just there for the case N<0, M<0 ?

if you meant the second modulo in the whole macro, i think it's necessary, because the expression for negative operands is missing the precision for positive ones, and just a simple modulo in C will solve this problem.

Reply:
I meant this modulo:
```
fmod(fmod(N, M) + M, M)
  ^
```
I think it is only needed to handle the case `N<0`, `M<0` when you don't use an XOR. As a call to `fmod` can be costly I think it would be better to use the XOR and only have one `fmod` call in each branch

Reply:
sorry, but i couldn’t follow in the XOR point, is the XOR will work if `N < 0` and `M < 0` ?.